### ISSUE TITLE
docs: hillshade-illumination-direction is documented as a closed range [0,359] but should actually be a half open range [0,360)

### ISSUE THREAD

**Issue by sargunv** [maplibre/maplibre-style-spec#939]
**Describe the bug**

(unsure if misleading documentation counts as a "bug" but it seems fitting)

The `hillshade-illumination-direction` property is documented as a closed range [0,359] but should actually be a half open range [0,360), because it's a floating point value; 359.5 is a valid value. I saw this while working on maplibre-compose and assumed it only accepts integer degrees, only to check the source code and realize it accepts floats.

Looking at the JSON style spec, I don't think it actually supports a way to define a half open range with a real number maximum, so it's not actually a trivial fix to the docs.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by HarelM** [comment]
Yup, the docs are generated from the v8 file, and the v8 content syntax is not great.
Feel free to try and heck it...
